sunpy/net/dataretriever/tests/test_swap.py

```diff
@@ -0,0 +1,58 @@
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+import datetime
+import pytest
+
+from sunpy.time.timerange import TimeRange
+from sunpy.net.vso.attrs import Time,Instrument,Level
+from sunpy.net.dataretriever.client import QueryResponse
+from sunpy.net.dataretriever.downloader_factory import UnifiedResponse
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+
+import sunpy.net.dataretriever.sources.swap as swap
+
+LCClient = swap.SWAPClient()
+
+@pytest.mark.online
+@pytest.mark.parametrize("timerange,url_start,url_end, level",
+                         [(TimeRange('2015/12/30 00:00:00','2015/12/30 23:59:59'),
+                           'http://proba2.oma.be/swap/data/bsd/2015/12/30/swap_lv1_20151230_000044.fits',
+                           'http://proba2.oma.be/swap/data/bsd/2015/12/30/swap_lv1_20151230_235935.fits', 1)])
+def test_get_url_for_time_range(timerange, url_start, url_end, level):
+    urls = LCClient._get_url_for_timerange(timerange, level = level)
+    assert isinstance(urls, list)
+    assert urls[0] == url_start
+    assert urls[-1] == url_end
+
+def test_can_handle_query():
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('swap'), a.Level(1)) == True
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('swap'), a.Level('q')) == True
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('swap'), a.Level('s')) == False
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('swap')) == False
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30','2015/12/31')) == False
+    assert swap.SWAPClient._can_handle_query(Time('2015/12/30 00:00:00','2015/12/31 00:05:00'), Instrument('eve')) == False
```

Comment:
I believe that David is correct to say that when testing that a function is meant to return True or False, then the test should use the 'is' idiom instead of the test for equality.  This is because we want to test that those functions really do return True or False, and not whatever Python considers is their equivalent.  For example

1 == True

does not cause an error in Python, but

1.0 == True
 
also evaluates to True.

In my opinion, if you expect that a function returns True or False, then the test should test for that explicitly.